src/test/unit/BLSSignatureCheckerUnit.t.sol

```diff
@@ -0,0 +1,229 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+import "../../contracts/middleware/BLSSignatureChecker.sol";
+import "../utils/BLSMockAVSDeployer.sol";
+
+contract BLSSignatureCheckerUnitTests is BLSMockAVSDeployer {
+    using BN254 for BN254.G1Point;
+
+    BLSSignatureChecker blsSignatureChecker;
+
+    function setUp() virtual public {
+        _setUpBLSMockAVSDeployer();
+
+        blsSignatureChecker = new BLSSignatureChecker(registryCoordinator);
+    }
+
+    function testBLSSignatureChecker_SingleQuorum_Valid(uint256 pseudoRandomNumber) public { 
+        uint256 numNonSigners = pseudoRandomNumber % (maxOperatorsToRegister - 1);
+
+        uint256 quorumBitmap = 1;
+        bytes memory quorumNumbers = BitmapUtils.bitmapToBytesArray(quorumBitmap);
+
+        (uint32 referenceBlockNumber, BLSSignatureChecker.NonSignerStakesAndSignature memory nonSignerStakesAndSignature) = 
+            _registerSignatoriesAndGetNonSignerStakeAndSignatureRandom(pseudoRandomNumber, numNonSigners, quorumBitmap);
+
+        uint256 gasBefore = gasleft();
+        blsSignatureChecker.checkSignatures(
+            msgHash, 
+            quorumNumbers,
+            referenceBlockNumber, 
+            nonSignerStakesAndSignature
+        );
+        uint256 gasAfter = gasleft();
+        emit log_named_uint("gasUsed", gasBefore - gasAfter);
+
+        // 0 nonSigners: 159908
+        // 1 nonSigner: 178683
+        // 2 nonSigners: 197410
```

Comment:
can we check anything here other than the call not reverting? it's a 'view' function so no state transition to check, but maybe we can check something about the returned values (looks like we're just tossing those out right now)

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/102/commits/db7e47e6a6b44022e7f3c6095441a357bf1588f2

Reply:
> [db7e47e](https://github.com/Layr-Labs/eigenlayer-contracts/commit/db7e47e6a6b44022e7f3c6095441a357bf1588f2)

looks like the commit you linked just puts the returned values into memory and then immediately dumps them? this is still short of checking anything about the returned values